core/libp2p/multi/utils/base58_codec.hpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_BASE58CODEC_HPP
+#define KAGOME_BASE58CODEC_HPP
+
+#include <array>
+#include <string>
+#include <gsl/span>
+#include <common/buffer.hpp>
+
+#include "outcome/outcome.hpp"
+
+
+
+namespace libp2p::multi {
+
+  /**
+   *  Base58 is a group of binary-to-text encoding schemes used to represent
+   * large integers as alphanumeric text
+   */
+  class Base58Codec {
```

Comment:
for base58 please reuse existing function from multibase

```
libp2p::multi::detail::encodeBase58
libp2p::multi::detail::decodeBase58
```

you can refactor expected::Result to outcome::result